src/AugmentedMatrix.jl

```diff
@@ -0,0 +1,62 @@
+using LinearAlgebra
+
+#Used in one line in function below
+function invert_or_zero(x :: Real)
+    if x == 0
+        return 0
+    else
+        return 1/x
+    end
+end
+
+#Code to compute augmented rate matrix
+function augmentedRateMatrix(list_of_rate_matrices :: Array{B}, list_of_time_steps :: Array{T}) where {T <: Real, B <: Matrix{T}}
+    number_of_times = length(list_of_time_steps)
+    @assert length(list_of_rate_matrices) == number_of_times
+    @assert length(list_of_rate_matrices) ≥ 1
+    number_of_states = size(list_of_rate_matrices[1])[1]
+
+    #Big in terms of memory.
+    J = zeros(number_of_states*number_of_times, number_of_states*number_of_times)
```

Comment:
Whenever the Type of the array's elements is known (in this case, we captured it as `T <: Real`, it's better to let `zeros` know the type by calling it as `zeros(T, ...)`.